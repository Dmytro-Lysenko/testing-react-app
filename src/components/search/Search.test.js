import { render, screen } from "@testing-library/react";
import useEvent from "@testing-library/user-event";

import Search from "./Search";

const onChange = jest.fn();

describe("Search component", () => {
  test(`renders Serach component`, () => {
    render(
      <Search value="" onChange={onChange}>
        Find :
      </Search>
    );

    expect(screen.getByText(/find/i)).toBeInTheDocument();
  });

  test(`renders without children`, () => {
    render(<Search value="" onChange={onChange} />);

    expect(screen.getByText(/search/i)).toBeInTheDocument();
  });

  test(`renders without placeholder`, () => {
    render(<Search value="" onChange={onChange} />);

    expect(screen.getByPlaceholderText(/search.../i)).toBeInTheDocument();
  });

  test(`custom placeholder work correctly`, () => {
    render(<Search value="" onChange={onChange} placeholder="find post" />);

    expect(screen.getByPlaceholderText(/find post/i)).toBeInTheDocument();
  });

//   test(`onChange works`, () => {
//     render(
//       <Search  value="" onChange={onChange}>
//         Find:
//       </Search>
//     );
//   });

//   useEvent.type(screen.getByRole('textbox'), 'React');

//   expect(onChange).toHaveBeenCalledTimes(5)

});
