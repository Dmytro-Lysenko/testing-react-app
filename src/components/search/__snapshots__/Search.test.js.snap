// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component Search snapshot 1`] = `undefined`;
